test/core/tsi/crl_ssl_transport_security_test.cc

```diff
@@ -54,103 +53,88 @@ const char* kSslTsiTestCrlSupportedCrlDirMissingIntermediate =
 const char* kSslTsiTestCrlSupportedCrlDirMissingRoot =
     "test/core/tsi/test_creds/crl_data/crls_missing_root/";
 const char* kSslTsiTestFaultyCrlsDir = "bad_path/";
+const std::string kRevokedKeyPath =
+    absl::StrCat(kSslTsiTestCrlSupportedCredentialsDir, "revoked.key");
+const std::string kRevokedCertPath =
+    absl::StrCat(kSslTsiTestCrlSupportedCredentialsDir, "revoked.pem");
+const std::string kValidKeyPath =
+    absl::StrCat(kSslTsiTestCrlSupportedCredentialsDir, "valid.key");
+const std::string kValidCertPath =
+    absl::StrCat(kSslTsiTestCrlSupportedCredentialsDir, "valid.pem");
+const std::string kRevokedIntermediateKeyPath = absl::StrCat(
+    kSslTsiTestCrlSupportedCredentialsDir, "leaf_signed_by_intermediate.key");
+const std::string kRevokedIntermediateCertPath = absl::StrCat(
+    kSslTsiTestCrlSupportedCredentialsDir, "leaf_and_intermediate_chain.pem");
+const std::string kRootCrlPath =
+    absl::StrCat(kSslTsiTestCrlSupportedCrlDir, "current.crl");
+const std::string kIntermediateCrlPath =
+    absl::StrCat(kSslTsiTestCrlSupportedCrlDir, "intermediate.crl");
+
+std::string LoadFile(absl::string_view file_path) {
+  grpc_slice slice;
+  GPR_ASSERT(grpc_load_file(file_path.data(), 1, &slice) == absl::OkStatus());
```

Comment:
Note that in the general case, it's not safe to assume that the string returned by `absl::string_view::data()` is null-terminated.  It may be that you are always passing in a null-terminated string to this function, but there's nothing in the API that guarantees this, so it seems like a bad choice of interface.  I suggest either changing the parameter from `absl::string_view` to `const char*` or changing this line to `std::string(file_path).c_str()`, which will make a copy and ensure that it is null-terminated.

Looks like this may have been a pre-existing bug.  And the question is moot anyway, as per my comment above, since this function should just go away.

Reply:
Now that this entire fn is gone, does this still hold for passing the file paths into `GetFileContents`?

Reply:
`GetFileContents()` takes a `char*` parameter instead of an `absl::string_view` parameter, so it's not an issue for that function.